libs/wingsdk/src/core/resource.ts

```diff
@@ -154,26 +154,30 @@ export abstract class Resource
       if (field.startsWith("this.")) {
         const key = field.substring(5);
         const obj: Resource = (this as any)[key];
-        if (!obj) {
+        if (obj === undefined) {
           throw new Error(
             `Resource ${this.node.path} does not have field ${key}`
           );
         }
-        if (!("_bind" in obj)) {
-          throw new Error(
-            `Resource ${this.node.path} field ${key} does not have a bind method`
-          );
-        }
-        obj._bind(host, resources[field]);
 
-        // add connection metadata
-        for (const op of resources[field]) {
```

Comment:
This PR adds a lot of duck-typing into the SDK for handling whether we're calling `_bind` or `_toInflight` on resources or objects. We should avoid this if possible since it doubles the number of cases a contributor/maintainer needs to think in many parts of the SDK code.

We can simplify this by annotating whether the captured item is a resource or a piece of immutable data. The compiler has this information, so we can just model that in the compiler's `Capture` struct and in the SDK's `InflightBinding` interface:

```rs
// capture.rs
struct Capture {
	pub object: Symbol,
	pub kind: CaptureType,
}

enum CaptureKind {
	Resource(CaptureDef),
	ImmutableData,
}
```

```ts
/**
 * A resource binding.
 */
export interface InflightBinding {
  /**
   * The resource. 
   */
  readonly resource?: IResource;
  
  /**
   * A piece of immutable data.
   */
  readonly object?: any;

  /**
   * The list of operations used on the resource.
   */
  readonly ops: string[];
}
```

Reply:
I'll look into it.

Alternatively I am tempted to explore the idea of basically implementing `toInflight` through `toJSON`. I have a gut feeling they are basically the same thing, but I need to think about it a little more.

Reply:
Done (as you suggested).

I still feel there's a potential opportunity for a unified model (of the universe), but for now, we'll just do with the general theory of relativity.